File: problems/704_binary_search/sol_2/main.c
int search(int* nums, int numsSize, int target) {
    int front = 0;
    int end = numsSize - 1;

    while(front <= end)
    {
        int mid = front + (end - front) / 2;

        if(nums[mid] == target)
            return mid;
        else if(nums[mid] < target)
            front = mid + 1;
        else
            end = mid - 1;
    }

    return -1;
}

File: problems/704_binary_search/sol_2/main.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        front = 0
        end = len(nums) - 1

        while front <= end:
            mid = front + (end - front) // 2

            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                front = mid + 1
            else:
                end = mid - 1
        
        return -1